contrib/epee/include/storages/portable_storage.h

```diff
@@ -241,40 +240,44 @@ namespace epee
     }
     //---------------------------------------------------------------------------------------------------------------
     template<class t_value>
-    bool portable_storage::set_value(const std::string& value_name, const t_value& v, hsection hparent_section)        
+    bool portable_storage::set_value(const boost::string_ref& value_name, const t_value& v, hsection hparent_section)        
     {
       BOOST_MPL_ASSERT(( boost::mpl::contains<boost::mpl::push_front<storage_entry::types, storage_entry>::type, t_value> )); 
       TRY_ENTRY();
       if(!hparent_section)
         hparent_section = &m_root;
-      storage_entry* pentry = find_storage_entry(value_name, hparent_section);
+      storage_entry* pentry = find_or_create_storage_entry(value_name, hparent_section);
       if(!pentry)
-      {
-        pentry = insert_new_entry_get_storage_entry(value_name, hparent_section, v);
-        if(!pentry)
-          return false;
-        return true;
-      }
+        return false;
       *pentry = storage_entry(v);
       return true;
       CATCH_ENTRY("portable_storage::template<>set_value", false);
     }
     //---------------------------------------------------------------------------------------------------------------
     inline
-    storage_entry* portable_storage::find_storage_entry(const std::string& pentry_name, hsection psection)
+    storage_entry* portable_storage::find_storage_entry(const boost::string_ref& pentry_name, hsection psection)
     {
       TRY_ENTRY();
       CHECK_AND_ASSERT(psection, nullptr);
-      auto it = psection->m_entries.find(pentry_name);
+      auto it = psection->m_entries.find(std::string(pentry_name.data(), pentry_name.size()));
```

Comment:
I assume the mention of C++14 in the commit was due to this function call here. `boost::container::map` has the same interface as the STL version, and has the templated find method available for C++03 or newer compilers. Not sure if its worth transitioning to the library, but it is there.

This interface (without the templated find) also seems preferable only if the majority of the sources are _not_ `std::string`. It seems the opposite is true, that ultimately all sources are `std::string`.